stm/asyncTaskQueue.mli

```diff
@@ -6,79 +6,129 @@
 (*         *       GNU Lesser General Public License Version 2.1        *)
 (************************************************************************)
 
+(** A status is
+    - `Fresh when XXX
+    - `Old x when XXX
+  *)
 type 'a worker_status = [ `Fresh | `Old of 'a ]
```

Comment:
A worker is `Fresh` when it borns, then when he ends a job he can either die (and be replaced by a fresh new worker) or hang there as an `Old` worker.  In such case some data can be carried by the `Old` constructor, tipycally used to implement `request_of_task`.

Reply:
I think one could get rid of `'a` and put competence there directly.

Reply:
Interesting, thanks; so is it correct to say that this API implements "one-shot" workers?

Reply:
No, it lets one implement both one-shot workers and "persistent" ones. E.g. par: is implement using workers that don't "reboot".  Proof workers do reboot mainly because the vm has some C state that cannot be cleared, so you have a real memory leak if you don't reboot the worker.